### ISSUE TITLE
search and filter throws errors

### ISSUE THREAD

**Issue by lublak** [nocodb/nocodb#2799]
**Please enter the following details**

```
Node: **v16.15.0**
Arch: **x64**
Platform: **win32**
Docker: **false**
Database: **pg**
ProjectOnRootDB: **false**
RootDB: **sqlite3**
PackageVersion: **0.92.4**
```

**Steps To Reproduce**

1. Search for dates: 2017-08-12 or text: ungültige Eingabesyntax für Typ bigint: »tes« (invalid inputsynta for typ bigint)
2. Filter: Same for filter (is equal) + is empty, is not empty

**Expected Behavior**

That the input value is adjusted to the respective value.
Example 2017-08-12, if the field is a date, it should be a date in the select, otherwise a number in a number and a text in a text.
The same applies to text. If the field is a number, it should be considered as "false" for is equal, etc.

In addition, I haven't even written out the date yet, for example, and you already get a red error message on the nose.

**Screenshots**

![grafik](https://user-images.githubusercontent.com/44057030/180231766-e1706409-cf57-42e4-a7a9-32d858c944b6.png)

**Comment by wingkwong** [comment]
This is my setup

<img width="517" alt="image" src="https://user-images.githubusercontent.com/35857179/187586335-e9b95d92-03d6-447b-a08d-5b47fc3a7452.png">

For Search for dates / texts, I'm not able to reproduce the issues.

<img width="591" alt="image" src="https://user-images.githubusercontent.com/35857179/187585771-c9c754e8-96b9-4481-988c-e868dfbe1d97.png">

<img width="519" alt="image" src="https://user-images.githubusercontent.com/35857179/187585800-a86f78cd-8ab7-465b-94e4-21950789deed.png">

When you search a column, have you selected the target column like below first?

<img width="219" alt="image" src="https://user-images.githubusercontent.com/35857179/187586407-c93ec58b-860f-4d73-b48d-afb120cdc544.png">

For Filter, I don't get what you mean. Do you mean you are using IsEqual and IsEmpty / isNotEmpty at the same time without search?

**Comment by lublak** [comment]
okay okay I see I made a mistake here myself because I didn't see this option:
![image](https://user-images.githubusercontent.com/44057030/187856259-f7eccb50-b00f-4461-a1ba-d14102b735f2.png)
I thought the search was across all fields. That was my mistake.
Apparently it was by default on the id field which is a number. If you then search here for text this red error message appears. Maybe it would make more sense (especially if I leave the user at it) to just show nothing because nothing was found.
@wingkwong 

And thanks for your answer :)

So to summarize again:
Select a field in the search with the data type Number. But search with text.
